iBudget/Entity/models.py

```diff
@@ -0,0 +1,82 @@
+from django.db import models
+from django.contrib.auth.models import (
+    BaseUserManager, AbstractBaseUser
+)
+
+
+class CustomUser(AbstractBaseUser):
+    email = models.EmailField(max_length=50, unique=True)
+    icon = models.CharField(max_length=30)
+
+    objects = BaseUserManager()
+    USERNAME_FIELD = 'email'
+
+
+class IncomeCategories(models.Model):
+    name = models.CharField(max_length=30)
+    owner = models.ForeignKey(CustomUser, on_delete=True)
+    date = models.DateField()
+    sum = models.DecimalField(max_digits=17, decimal_places=2)
+    icon = models.CharField(max_length=30)
+
+
+class FundsCategories(models.Model):
+    name = models.CharField(max_length=30)
+    icon = models.CharField(max_length=30)
+    is_shared = models.BooleanField(default=False)
+    owner = models.ForeignKey(CustomUser, on_delete=True)
+
+
+class SpendingCategories(models.Model):
+    name = models.CharField(max_length=30)
+    owner = models.ForeignKey(CustomUser, on_delete=True)
+    icon = models.CharField(max_length=30)
+    is_shared = models.BooleanField(default=False)
+
+
+class Group(models.Model):
+    name = models.CharField(max_length=30)
+    icon = models.CharField(max_length=30)
+    owner = models.ForeignKey(CustomUser, on_delete=False)
+    members = models.ManyToManyField(CustomUser,
+                                     through='UsersInGroups',
+                                     related_name="groups")
+    shared_funds = models.ManyToManyField(FundsCategories,
+                                          through='SharedFunds',
+                                          related_name="groups")
+    shared_spendings = models.ManyToManyField(SpendingCategories,
+                                              through='SharedSpendingCategories',
+                                              related_name="groups")
+
+
+class IncomeHistory(models.Model):
+    income = models.ForeignKey(IncomeCategories, on_delete=True, related_name="income_history")
+    funds = models.ForeignKey(FundsCategories, on_delete=True, related_name="income_history")
```

Comment:
`funds` was described as `ForeignKey` so maybe it should be named in singular.

Reply:
Ok